ido/src/lib.rs

```diff
@@ -254,6 +253,51 @@ impl IDOContract {
         self.projects.insert(&project_id, &project);
     }
 
+    
+    pub fn commit(&mut self,account_id:AccountId, project_id: ProjectId, fund_contract_id: AccountId, deposit: U128)-> Promise{
+        let project = self.internal_get_project_or_panic(project_id);
+        if project.fund_contract_id == fund_contract_id{
+            let buy_amount = self.internal_sale_commit(project_id,&account_id,deposit.0);
+            
+            env::log(format!("Total buyed: {}",buy_amount ).as_bytes());
+            if deposit.0 > buy_amount {
+                ext_ft_contract::ft_transfer(
+                    account_id.clone(), 
+                    U128(deposit.0 - buy_amount), 
+                    Some(format!("Transfer change to signer")), 
+                    &fund_contract_id, 
+                    DEPOSIT_ONE_YOCTOR, 
+                    FT_TRANSFER_GAS).then(
+                        ext_ft_contract::ft_transfer(
+                            project.fund_contract_id, 
+                            U128(buy_amount), 
+                            Some(format!("Transfer {} from {}",buy_amount,account_id.clone())), 
+                            &fund_contract_id, 
+                            DEPOSIT_ONE_YOCTOR, 
+                            FT_TRANSFER_GAS)
+                    )
+            } else {
+                ext_ft_contract::ft_transfer(
```

Comment:
We don't immediately transfer the deposit amount of the ft token of the user to the project owner for two reasons:
- The user can claim it back in the distribution period if if is still remaining.
- If we do this transfer in each deposit transaction that will increase the gas fee and the user must pay for that. It's necessary. The project owner can claim all funding balance once after the sale period is end. 

Also if we do this, the receiver must be `project.owner_id`, not `project.fund_contract_id'.